src/Uno.UI/UI/Xaml/Automation/AutomationProperties.cs

```diff
@@ -148,6 +149,13 @@ private static void OnAutomationIdChanged(DependencyObject dependencyObject, Dep
 			if (FrameworkElementHelper.IsUiAutomationMappingEnabled && dependencyObject is UIElement uiElement)
 			{
 				uiElement.SetAttribute("xamlautomationid", (string)args.NewValue);
+				uiElement.SetAttribute("aria-label", (string)args.NewValue);
+
+				var role = FindHtmlRole(uiElement);
+				if (role != null)
+				{
+					uiElement.SetAttribute("role", role);
+				}
```

Comment:
Please use the overload of `.SetAttribute()` to call it only once.  Interop with javascript are costly.